DataManagementSystem/Agent/RequestOperations/PutAndRegister.py

```diff
@@ -112,73 +161,113 @@ def __call__( self ):
       # # call DataManager passing a list of requested catalogs
       catalogs = self.operation.Catalog
       if catalogs:
-        catalogs = [ cat.strip() for cat in catalogs.split( ',' ) ]
-      putAndRegister = DataManager( catalogs = catalogs ).putAndRegister( lfn,
-                                                                          pfn,
-                                                                          targetSE,
-                                                                          guid = guid,
-                                                                          checksum = checksum )
+        catalogs = [cat.strip() for cat in catalogs.split(',')]
+      putAndRegister = DataManager(catalogs=catalogs).putAndRegister(lfn,
+                                                                     pfn,
+                                                                     targetSE,
+                                                                     guid=guid,
+                                                                     checksum=checksum)
+      # Template object for a failing file object.
+      if self.rmsMonitoring:
+        failFileObj = {
+            "host": Network.getFQDN(),
+            "objectType": "File",
+            "operationType": self.operation.Type,
+            "parentID": self.operation.OperationID,
+            "status": "Failed",
+            "nbObject": 1
+        }
       if not putAndRegister["OK"]:
-        gMonitor.addMark( "PutFail", 1 )
+        if self.rmsMonitoring:
+          failFileObj["timestamp"] = int(Time.toEpoch())
+          self.rmsMonitoringReporter.addRecord(failFileObj)
```

Comment:
that can't possibly work.
You are constantly adding the same dictionary and updating the field, so in practice you are sending multiple time the exact same info.